app/models/cdr/cdr.rb

```diff
@@ -185,8 +183,10 @@ def status_sym
   end
 
 
+  # TODO: use DisconnectInitiator#display_name instead
+  # Why in was cached this way? `DISCONNECTORS = DisconnectInitiator.all.index_by(&:id)`
   def disconnect_initiator_name
-    DISCONNECTORS[self.disconnect_initiator_id].try(:name)
+    disconnect_initiator.try(:name)
```

Comment:
@workgena @dmitry-sinina  maybe we can use constant as we did before, I think we can make it lazy if we need, 

http://greyblake.com/blog/2014/10/05/lazy-object-pattern-in-ruby/
```
class LazyObject < ::BasicObject
  def initialize(&callable)
    @callable = callable
  end

  def __target_object__
    @__target_object__ ||= @callable.call
  end

  def method_missing(method_name, *args, &block)
    __target_object__.send(method_name, *args, &block)
  end
end


```

and then we can 

```
DISCONNECTORS = LazyObject.new { DisconnectInitiator.all.index_by(&:id) } 
```